plugins/out_stackdriver/stackdriver.c

```diff
@@ -2596,8 +2616,11 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
     update_retry_metric(ctx, event_chunk, ts, c->resp.status, ret_code);
 #endif
 
+
     /* Cleanup */
-    flb_sds_destroy(payload_buf);
+    if (out_payload_buf != payload_buf) {
+        flb_free(out_payload_buf);
```

Comment:
This is related to what I previously mentioned and to make things clearer it'd be better if you renamed the variable to something like `compressed_payload_buffer` which would clearly convey its purpose.

Reply:
Thanks! Done.